packages/data-stores/src/plans/selectors.ts

```diff
@@ -10,11 +10,9 @@ function getPlan( slug: PlanSlug ) {
 	return PLANS_LIST[ slug ];
 }
 
-export const getSelectedPlan = ( state: State ) =>
-	state.selectedPlanSlug ? getPlan( state.selectedPlanSlug ) : null;
+export const getPlanBySlug = ( _: State, slug: PlanSlug ) => getPlan( slug );
 
-export const getDefaultPlan = ( _: State, hasPaidDomain: boolean, hasPaidDesign: boolean ) =>
```

Comment:
`plans` store shouldn't care about external state such as `hasPaidDomain` or `hasPaidDesign`.

Reply:
Thanks. I can see that now `useSelectedPlan()`, a hook in the consumer app, now determines what plan to return.  If a free plan is selected, but a paid domain or paid design is selected , it will change to a paid plan. Is there a place where `updatePlan()` is called when this is changed?

Reply:
> If a free plan is selected, but a paid domain or paid design is selected

At the moment, only if a paid domain is selected because we still don't have the UI ready to connect the concepts of paid plan <=> premium design. Check this convo: https://github.com/Automattic/wp-calypso/issues/42601#issuecomment-644742819

> Is there a place where setPlan()/updatePlan() is called when this is changed or is this not necessary?

We're not updating the selected plan in these cases in Gutenboarding because we always want to show Plans grid. See in code: https://github.com/Automattic/wp-calypso/pull/43935/files#diff-7ccae80646b579bcdfe01b44b7601d43R57-R59